test/SlowTests/Authentication/AuthenticationClusterTests.cs

```diff
@@ -104,5 +115,271 @@ public async Task CanReplaceClusterCert()
+
+                var mre = new ManualResetEventSlim();
+
+                leader1.ServerCertificateChanged += (sender, args) => mre.Set();
+
+                var requestExecutor = store1.GetRequestExecutor();
+                using (requestExecutor.ContextPool.AllocateOperationContext(out JsonOperationContext context))
+                {
+                    var command = new ReplaceClusterCertificateOperation(certBytes, false)
+                        .GetCommand(store1.Conventions, context);
+
+                    requestExecutor.Execute(command, context);
+                }
+
+                Assert.True(mre.Wait(Debugger.IsAttached ? TimeSpan.FromMinutes(10) : TimeSpan.FromMinutes(2)), "Waited too long");
+                Assert.NotNull(leader1.Certificate.Certificate.Thumbprint);
+                Assert.True(leader1.Certificate.Certificate.Thumbprint.Equals(cluster1ReplacementCert.Thumbprint), "New cert is identical");
+
+                // Disable external replication
+                var external = new ExternalReplication(store1.Database, $"ConnectionString-{store2.Identifier}")
+                {
+                    TaskId = externalList.First().TaskId,
+                    Disabled = true
+                };
+
+                var responsibleNode = externalList[0].ResponsibleNode;
+
+                var clusterNodes1 = Servers.Where(s => s.ServerStore.GetClusterTopology().TryGetNodeTagByUrl(leader1.WebUrl).HasUrl);
+                var node = clusterNodes1.Single(n => n.ServerStore.NodeTag == responsibleNode);
+
+                var db1 = await node.ServerStore.DatabasesLandlord.TryGetOrCreateResourceStore(store1.Database);
+                Assert.NotNull(db1);
+                var replicationConnection = db1.ReplicationLoader.OutgoingHandlers.Single(x => x.Destination is ExternalReplication);
+                var res = await store1.Maintenance.SendAsync(new UpdateExternalReplicationOperation(external));
+
+                Assert.Equal(externalList.First().TaskId, res.TaskId);
+
+                // Make sure the command is processed
+                await db1.ServerStore.Cluster.WaitForIndexNotification(res.RaftCommandIndex);
+                var connectionDropped = await WaitForValueAsync(() => replicationConnection.IsConnectionDisposed, true);
+                Assert.True(connectionDropped);
+
+                // Enable external replication
+                external.Disabled = false;
+                res = await store1.Maintenance.SendAsync(new UpdateExternalReplicationOperation(external));
+                Assert.Equal(externalList.First().TaskId, res.TaskId);
+
+                using (var session = store1.OpenAsyncSession())
+                {
+                    await session.StoreAsync(new User { Name = "Avivush" }, "users/2");
+                    await session.SaveChangesAsync();
+                }
+
+                var replicated = WaitForDocumentToReplicate<User>(store2, "users/2", 10000);
```

Comment:
Check for replication _errors_, not wait for this

Reply:
Currently auth errors aren't caught here. Opened a separate issue: https://issues.hibernatingrhinos.com/issue/RavenDB-13010